### PR TITLE
Better exception behavior

### PR DESCRIPTION
**Pull request by mattdrees**
This improves behavior in two ways:

  * better datadog error metrics
  * better structuring of context (esp. report name), so I can eventually write better rollbar fingerprinting rules.

### CODE REVIEW

File: src/main/java/org/ccci/obiee/client/rowmap/impl/AnalyticsManagerFactoryImpl.java [hunk 1]
```diff
@@ -106,6 +110,10 @@ public AnalyticsManager createAnalyticsManager()
                 converterStore,
                 tracer
             );
+        } catch (Exception e){
```
Comment (wrandall22) [inline src/main/java/org/ccci/obiee/client/rowmap/impl/AnalyticsManagerFactoryImpl.java]: Nitpicky, but maybe put a space before the `{`

File: src/main/java/org/ccci/obiee/client/rowmap/DataRetrievalException.java [hunk 2]
```diff
@@ -8,16 +17,96 @@
...
+            return null;
+        }
+    }
+
+    private Map<String, Object> formatParamsAsString(ReportParams params)
+    {
+        return params == null ? null : asMap(params.getVariables());
+    }
+
+    private Map<String, Object> asMap(List<Variable> variables)
```
Comment (wrandall22) [inline src/main/java/org/ccci/obiee/client/rowmap/DataRetrievalException.java]: This project is using Java 8. You could use streams, something like this:
```
return variables.stream().collect(
  Collectors.toMap(variable -> variable.getName(), variable -> variable.getValue()));
```

File: src/main/java/org/ccci/obiee/client/rowmap/impl/AnalyticsManagerFactoryImpl.java (hunk 1 above)
Comment (mattdrees) [reply src/main/java/org/ccci/obiee/client/rowmap/impl/AnalyticsManagerFactoryImpl.java]: I don't mind nitpicks

**Comment by mattdrees** [comment]
thanks!

**Pull request merged**